src/analysis/inlay_hints.ml

```diff
@@ -0,0 +1,183 @@
+open Std
+
+let {Logger.log} = Logger.for_section "inlay-hints"
+
+module Iterator = Ocaml_typing.Tast_iterator
+
+let overlap_with_loc (start, stop) loc =
+  let a = Lexing.compare_pos start loc.Location.loc_end
+  and b = Lexing.compare_pos stop loc.Location.loc_start in
+  a <= 0 && b >= 0 || a >= 0 && b <= 0
```

Comment:
I have the usual question: doesn't this helper function already exists elsewhere ? If not, maybe it could :-)

Reply:
Right, moved into `Location_aux`.